backend/persistent/datastores/drivers/AWSAthenaDriver.js

```diff
@@ -0,0 +1,222 @@
+            if (err) {
+                return reject(err);
+            }
+            return resolve(data);
+        });
+    });
+}
+
+/**
+ * The following function will execute a query against athena.  It will first start
+ * by starting the query request.  It will then start a timer to periodically
+ * check to see if the query results have completed. If the Query Timeout has
+ * Exceeded it will reject the query.  If the it receives data before the query
+ * times out it will return the response.
+ * @param {object} queryParams - Query Parameters
+ * @param {string} queryParams.accessKey - AWS Access Key
+ * @param {string} queryParams.secretAccessKey - AWS Secret Key
+ * @param {string} queryParams.region - AWS Region
+ * @param {string} queryParams.region - AWS Region
+ * @param {string} queryParams.dbName - Database name to connect to
+ * @param {string} queryParams.sqlStatement - SQL statement to execute
+ * @param {string} queryParams.s3Outputlocation - Location will Athena will output resutls of query
+ * @param {number} queryParams.queryTimeout  - The timeout interval when the query should stop
+ * @returns {Promise} resolve to AWS Query Response
+ */
+export function executeQuery(queryParams) {
+    const client = createAthenaClient(queryParams);
+
+    return new Promise(function(resolve, reject) {
+        return startQuery(client, queryParams).then(queryExecutionId => {
+
+            // Define the wait interval
+            let retryInterval = queryParams.queryTimeout / NUMBER_OF_RETRIES;
+
+            let retryCount = 0;
+            // If retry interval is not defined or less 0 set retry to 1000
+            if ((!retryInterval) || (retryInterval < 1)) {
+                retryInterval = 1000;
+            }
+
+            const checkQueryStatus = () => {
+                retryCount++;
+                queryResultsCompleted(client, queryExecutionId).then(queryStatus => {
+                    if (queryStatus < 0) {
+                        return reject('There was an error completing the query');
+                    } else if (queryStatus === 1) {
+                        return getQueryResults(client, queryExecutionId).then(rst => {
+
+                            if (rst && rst.ResultSet && rst.ResultSet.Rows) {
+                                return resolve(rst.ResultSet.Rows);
+                            }
+                                return resolve([]);
+                        });
+                    } else if (retryCount > NUMBER_OF_RETRIES) {
+                        return reject('Timeout.  Athena did not respond before the query timeout.');
+                    }
+                        // Loop again
+                        return setTimeout(checkQueryStatus, retryInterval);
+
+                }).catch(err => {
+                    return reject(err);
+                });
+            };
+
+            checkQueryStatus();
+
+        }).catch(err => {
+            return reject(err);
```

Comment:
😄 I'm surprised there is no `Logger.log` here, but since we'll remove them eventually, no need to add it.